test/e2e/multiple_secret_versions_autorotation_test.go

```diff
@@ -0,0 +1,499 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"html/template"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/pkg/provider"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/exec"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/helm"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/namespace"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/pod"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/secret"
+	"github.com/Azure/secrets-store-csi-driver-provider-azure/test/e2e/framework/secretproviderclass"
+
+	"github.com/ghodss/yaml"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	"sigs.k8s.io/secrets-store-csi-driver/apis/v1alpha1"
+)
+
+const (
+	multipleSecretsRandomLength = 5
+)
+
+var _ = Describe("Test auto rotation of mount contents and K8s secrets", func() {
+	var (
+		specName = "multiversionautorotation"
+		spc      *v1alpha1.SecretProviderClass
+		ns       *corev1.Namespace
+		p        *corev1.Pod
+	)
+
+	BeforeEach(func() {
+		ns = namespace.Create(namespace.CreateInput{
+			Creator: kubeClient,
+			Name:    specName,
+		})
+	})
+
+	AfterEach(func() {
+		Cleanup(CleanupInput{
+			Namespace: ns,
+			Getter:    kubeClient,
+			Lister:    kubeClient,
+			Deleter:   kubeClient,
+		})
+	})
+
+	It("should auto rotate mount contents with service principal", func() {
+		if config.IsKindCluster {
+			Skip("test case not supported for kind cluster")
+		}
+
+		nodePublishSecretRef := secret.Create(secret.CreateInput{
+			Creator:   kubeClient,
+			Name:      "secrets-store-creds",
+			Namespace: ns.Name,
+			Data:      map[string][]byte{"clientid": []byte(config.AzureClientID), "clientsecret": []byte(config.AzureClientSecret)},
+			Labels:    map[string]string{"secrets-store.csi.k8s.io/used": "true"},
+		})
+
+		secretName := fmt.Sprintf("secret-sp-%s", utilrand.String(multipleSecretsRandomLength))
+		// create secret in keyvault
+		err := kvClient.SetSecret(secretName, "secret")
+		Expect(err).To(BeNil())
+		defer func() {
+			err = kvClient.DeleteSecret(secretName)
+			Expect(err).To(BeNil())
+		}()
+
+		keyVaultObjects := []provider.KeyVaultObject{
+			{
+				ObjectName:           secretName,
+				ObjectType:           provider.VaultObjectTypeSecret,
+				ObjectVersionHistory: 5,
+			},
+		}
```

Comment:
We don't have secrets with multiple versions setup for CI. We can do this after we hash out the feedback in this PR.